File: install-firetiger/assets/nodejs/instrumentation.ts
import { NodeSDK } from '@opentelemetry/sdk-node';
import { getNodeAutoInstrumentations } from '@opentelemetry/auto-instrumentations-node';
import { OTLPTraceExporter } from '@opentelemetry/exporter-trace-otlp-http';
import { OTLPMetricExporter } from '@opentelemetry/exporter-metrics-otlp-http';
import { OTLPLogExporter } from '@opentelemetry/exporter-logs-otlp-http';
import { PeriodicExportingMetricReader } from '@opentelemetry/sdk-metrics';
import { BatchLogRecordProcessor } from '@opentelemetry/sdk-logs';
import { resourceFromAttributes } from '@opentelemetry/resources';
import { ATTR_SERVICE_NAME } from '@opentelemetry/semantic-conventions';

const OTEL_ENDPOINT = process.env.OTEL_EXPORTER_OTLP_ENDPOINT || '{{INGEST_URL}}';
const OTEL_AUTH = process.env.OTEL_EXPORTER_OTLP_HEADERS?.match(/Basic\s+(.+)/)?.[1]
  ?? '{{AUTH_HEADER}}';
const OTEL_HEADERS = { Authorization: `Basic ${OTEL_AUTH}` };

const resource = resourceFromAttributes({
  [ATTR_SERVICE_NAME]: process.env.OTEL_SERVICE_NAME || '{{SERVICE_NAME}}',
});

const sdk = new NodeSDK({
  resource,
  traceExporter: new OTLPTraceExporter({
    url: `${OTEL_ENDPOINT}/v1/traces`,
    headers: OTEL_HEADERS,
  }),
  metricReader: new PeriodicExportingMetricReader({
    exporter: new OTLPMetricExporter({
      url: `${OTEL_ENDPOINT}/v1/metrics`,
      headers: OTEL_HEADERS,
    }),
  }),
  logRecordProcessor: new BatchLogRecordProcessor(
    new OTLPLogExporter({
      url: `${OTEL_ENDPOINT}/v1/logs`,
      headers: OTEL_HEADERS,
    })
  ),
  instrumentations: [getNodeAutoInstrumentations()],
});

sdk.start();

process.on('SIGTERM', () => {
  sdk.shutdown()
    .then(() => process.exit(0))
    .catch(() => process.exit(1));
});

File: install-firetiger/assets/python/instrumentation.py
import os
import logging
from opentelemetry import trace, metrics
from opentelemetry.sdk.trace import TracerProvider
from opentelemetry.sdk.trace.export import BatchSpanProcessor
from opentelemetry.sdk.metrics import MeterProvider
from opentelemetry.sdk.metrics.export import PeriodicExportingMetricReader
from opentelemetry.sdk.resources import Resource, SERVICE_NAME
from opentelemetry.exporter.otlp.proto.http.trace_exporter import OTLPSpanExporter
from opentelemetry.exporter.otlp.proto.http.metric_exporter import OTLPMetricExporter
from opentelemetry._logs import set_logger_provider
from opentelemetry.sdk._logs import LoggerProvider, LoggingHandler
from opentelemetry.sdk._logs.export import BatchLogRecordProcessor
from opentelemetry.exporter.otlp.proto.http._log_exporter import OTLPLogExporter

OTEL_ENDPOINT = os.getenv('OTEL_EXPORTER_OTLP_ENDPOINT', '{{INGEST_URL}}')
OTEL_AUTH = os.getenv('OTEL_AUTH_HEADER', 'Basic {{AUTH_HEADER}}')
OTEL_HEADERS = {'Authorization': OTEL_AUTH}
SERVICE = os.getenv('OTEL_SERVICE_NAME', '{{SERVICE_NAME}}')

resource = Resource.create({SERVICE_NAME: SERVICE})

# Traces
_tracer_provider = TracerProvider(resource=resource)
_tracer_provider.add_span_processor(
    BatchSpanProcessor(
        OTLPSpanExporter(
            endpoint=f"{OTEL_ENDPOINT}/v1/traces",
            headers=OTEL_HEADERS,
        )
    )
)
trace.set_tracer_provider(_tracer_provider)

# Metrics
_metric_reader = PeriodicExportingMetricReader(
    OTLPMetricExporter(
        endpoint=f"{OTEL_ENDPOINT}/v1/metrics",
        headers=OTEL_HEADERS,
    )
)
_meter_provider = MeterProvider(resource=resource, metric_readers=[_metric_reader])
metrics.set_meter_provider(_meter_provider)

# Logs
_logger_provider = LoggerProvider(resource=resource)
_logger_provider.add_log_record_processor(
    BatchLogRecordProcessor(
        OTLPLogExporter(
            endpoint=f"{OTEL_ENDPOINT}/v1/logs",
            headers=OTEL_HEADERS,
        )
    )
)
set_logger_provider(_logger_provider)
logging.getLogger().addHandler(LoggingHandler(logger_provider=_logger_provider))

# Convenience exports
tracer = trace.get_tracer(__name__)
meter = metrics.get_meter(__name__)
